### PR TITLE
RESTEasy on Vert.x by default

### PR DESCRIPTION
**Pull request by patriot1burke**
* quarkus-resteasy no longer pulls in undertow. It instead runs directly on Vert.x Web
* To run resteasy within a servlet container just add quarkus-undertow to pom dependencies
* Adds vertx StaticHandler for META-INF/resources files.
* Keeps exception mappers

### CODE REVIEW

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxBlockingOutput.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    protected final HttpServerRequest request;
+
+    public VertxBlockingOutput(HttpServerRequest request) {
+        this.request = request;
+        request.response().exceptionHandler(new Handler<Throwable>() {
+            @Override
+            public void handle(Throwable event) {
+                //log.debugf(event, "IO Exception ");
+                //TODO: do we need this?
+                event.printStackTrace();
```
Comment (stuartwdouglas) [inline extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxBlockingOutput.java]: This should use a logger

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]

File: extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java [hunk 2]
```diff
@@ -370,20 +376,25 @@ JaxbEnabledBuildItem enableJaxb() {
...
+    private static void registerProviders(ResteasyDeployment deployment,
+            Map<String, String> resteasyInitParameters,
             BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             BuildProducer<UnremovableBeanBuildItem> unremovableBeans,
             JaxrsProvidersToRegisterBuildItem jaxrsProvidersToRegisterBuildItem) {
 
         if (jaxrsProvidersToRegisterBuildItem.useBuiltIn()) {
             // if we find a wildcard media type, we just use the built-in providers
             resteasyInitParameters.put(ResteasyContextParameters.RESTEASY_USE_BUILTIN_PROVIDERS, "true");
+            deployment.setRegisterBuiltin(true);
```
[the remark this answers is not in the document]
Comment (patriot1burke) [reply extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java]: Will save this for another day, ok?

[the remark this answers is not in the document]
Comment (patriot1burke) [reply extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java]: I'm just not sure if resteasy requires refactoring to support this.

**Comment by stuartwdouglas** [comment]
I will see what others think, but I want to fork a 0.23 branch now and merge this ASAP so we can start to make progress

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**